src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -120,6 +124,24 @@ public object GetCursor(IStreamIdentity streamIdentity, StreamSequenceToken sequ
             return cursor;
         }
 
+        private void ReportCacheMessageStatistics(object state)
+        {
+            if (this.IsEmpty)
+            {
+                this.cacheMonitor.ReportMessageStatistics(null, null, null, this.ItemCount);
```

Comment:
Fair enough.  I also made that comment before I reviewed the ItemCount logic.  I thought it actually tried to do work (like it used to).